lib/gui/os/dropzone/directives/dropzone.js

```diff
@@ -42,17 +41,39 @@ module.exports = ($timeout) => {
     },
     link: (scope, element) => {
       const domElement = element[0];
+      let hoverLayer = 0;
```

Comment:
This counter always looked strange to me. I've investigated in more depth and I can definitely understand the reason now: the drag events seem to go nuts even when you stay over the overlay, which is definitely not expected behaviour.

I did found a fix though. If you set `pointer-events: none;` to the **direct children** of the element containing the `os-dropzone` directive then suddenly the events starts behaving normally, and you can reduce this to:

```javascript
domElement.ondragleave = (event) => {
  event.preventDefault();
  event.currentTarget.classList.remove('dropzone-hover');
};

domElement.ondragend = (event) => {
  event.preventDefault();
  event.currentTarget.classList.remove('dropzone-hover');
};

domElement.ondragenter = (event) => {
  event.preventDefault();
  event.currentTarget.classList.add('dropzone-hover');
};

domElement.ondrop = (event) => {
  event.preventDefault();
  const filename = event.dataTransfer.files[0].path;
  event.currentTarget.classList.remove('dropzone-hover');
```

Which means you don't even need a mutable variable at all. I guess we can make the directive add that CSS rule to its children on drag enter, and remove it when done.

As a bonus, can you extract the `dropzone-hover` string as a constant?